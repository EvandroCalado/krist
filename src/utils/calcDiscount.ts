export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
