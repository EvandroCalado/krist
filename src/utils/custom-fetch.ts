import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
