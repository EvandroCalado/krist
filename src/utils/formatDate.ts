export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
