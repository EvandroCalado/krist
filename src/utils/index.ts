export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
