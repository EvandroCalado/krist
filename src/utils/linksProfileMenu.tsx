import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
