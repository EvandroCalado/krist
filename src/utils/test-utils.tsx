import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
