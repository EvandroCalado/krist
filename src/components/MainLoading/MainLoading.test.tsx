import { screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { MainLoading } from '.';

describe('MainLoading', () => {
  it('should render default', () => {
    customRender(<MainLoading />);

    const svg = screen.getByTestId('svg');
    const span = screen.getByText('Carregando...');

    expect(svg).toBeInTheDocument();
    expect(span).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = customRender(<MainLoading />);

    expect(container).toMatchSnapshot();
  });
});
