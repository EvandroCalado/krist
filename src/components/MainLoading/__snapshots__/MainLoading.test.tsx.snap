// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLoading should render a snapshot 1`] = `
.c0 {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 2.4rem;
}

.c0>svg {
  stroke-width: 1px;
  stroke-dasharray: 360;
  stroke-dashoffset: 1000;
  animation: draw 2s ease-in-out;
}

.c1 {
  text-transform: uppercase;
  animation: blink 800ms ease-in-out infinite;
}

<div>
  <div
    class="c0"
  >
    <svg
      data-testid="svg"
      height="100"
      id="SVGRoot"
      version="1.1"
      viewBox="0 0 30 30"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        stroke="#000"
        stroke-width="1px"
        transform="matrix(.5277 0 0 .52555 .36018 -5.0307)"
      >
        <path
          d="m-0.09638 10.718 12.51 10.912-12.51 9.7809z"
        />
        <path
          d="m0.86201 37.182 35.08-27.046 19 9.66e-4 -32.121 27.035z"
        />
        <path
          d="m-0.096472 45.676 12.352 9.8645-12.347 10.088z"
        />
        <path
          d="m0.90032 40 21.927-6e-3 32.083 26.197-18.915-0.0079z"
        />
      </g>
    </svg>
    <span
      class="c1"
    >
      Carregando...
    </span>
  </div>
</div>
`;
